pkg/scheduler/factory/factory.go

```diff
@@ -752,8 +752,17 @@ func (c *configFactory) onServiceAdd(obj interface{}) {
 func (c *configFactory) onServiceUpdate(oldObj interface{}, newObj interface{}) {
 	if c.enableEquivalenceClassCache {
 		// TODO(resouer) We may need to invalidate this for specified group of pods only
-		oldService := oldObj.(*v1.Service)
-		newService := newObj.(*v1.Service)
+		newService, ok := newObj.(*v1.Service)
```

Comment:
are there expected code paths that can lead to this? if so, is logging and ignoring the correct thing to do?

Reply:
It should never happen.

Reply:
I think it's better to consider the function itself, not whether it will happen or not.

Reply:
> I think it's better to consider the function itself, not whether it will happen or not.

If encountering this indicates a programming error, is logging it in a way that will probably be ignored what we want to do?

Reply:
That's a good question.
I think it doesn't really hurt, because if we got something that is not a service, then we can't really do much with it. OTOH it indicates programming error.

I don't have strong opinion here. @liggitt - do you prefer to leave it as is?

Reply:
typically for things that indicate programming errors, and are in controller-like components, we use these as assertions, rather than just logging

Reply:
Fine - let's close it then.